platform-intel.c

```diff
@@ -669,14 +724,251 @@ static int read_efi_variable(void *buffer, ssize_t buf_size,
+ * In case of success, the structure &uefi_acpi_table will be filled.
+ *
+ * Return: %MDADM_STATUS_SUCCESS on success, %MDADM_STATUS_ERROR otherwise.
+ */
+mdadm_status_t read_uefi_acpi_table(int fd, char *file_name, struct uefi_acpi_table *table)
+{
+	assert(is_fd_valid(fd));
+
+	if (read(fd, table, sizeof(struct uefi_acpi_table)) == -1) {
+		pr_err("Fail to read ACPI UEFI file. File: %s, Error: %s\n",
+		       file_name, strerror(errno));
+		return MDADM_STATUS_ERROR;
+	}
+	return MDADM_STATUS_SUCCESS;
+}
+
+/**
+ * read_uefi_acpi_orom_data() - read OROM data from UEFI ACPI table.
+ * @fd: file descriptor.
+ * @file_name: name of ACPI table.
+ * @uefi_table: struct to fill out.
+ *
+ * Read OROM from ACPI UEFI table under given file descriptor.
+ * Table must have the appropriate OROM data, which should be confirmed before call this function.
+ * In case of success, &orom in structure in &uefi_table will be filled..
+ *
+ * Return: %MDADM_STATUS_SUCCESS on success, %MDADM_STATUS_ERROR otherwise.
+ */
+mdadm_status_t
+read_uefi_acpi_orom_data(int fd, char *file_name, uefi_acpi_table_with_orom_t *uefi_table)
+{
+	assert(is_fd_valid(fd));
+
+	// set data offset for OROM read
+	if (lseek(fd, uefi_table->table.dataOffset, 0) == -1L) {
+		pr_err("Fail to reposition offset in ACPI UEFI file. File: %s, Error: %s\n",
+		       file_name, strerror(errno));
+		return MDADM_STATUS_ERROR;
+	}
+
+	if (read(fd, &uefi_table->orom, sizeof(uefi_table->orom)) == -1) {
+		pr_err("Fail to read OROM data from ACPI UEFI file. File: %s, Error: %s\n",
+		       file_name, strerror(errno));
+		return MDADM_STATUS_ERROR;
+	}
+
+	return MDADM_STATUS_SUCCESS;
+}
+
+/**
+ * verify_uefi_acpi_table_length() - verify if ACPI UEFI table have correct length with focus at
+ * OROM.
+ * @table: struct with UEFI table.
+ *
+ * Verify if ACPI UEFI table have correct length with focus at OROM. Make sure that the file is
+ * correct and contains the appropriate length data based on the length of the OROM.
+ *
+ * Return: %true if length is correct, %false otherwise.
+ */
+bool verify_uefi_acpi_table_length(struct uefi_acpi_table table)
+{
+	if (table.length < ACPI_UEFI_DATA_OFFSET)
+		return false;
+
+	if (table.length <= table.dataOffset)
+		return false;
+
+	if (table.length - table.dataOffset != sizeof(struct imsm_orom))
+		return false;
+	return true;
+}
+
+/**
+ * find_orom_in_acpi_uefi_tables() - find OROM in UEFI ACPI tables based on requested controllers.
+ * @uefi_controllers: array of requested UEFI controllers.
+ * @uefi_controllers_number: number of UEFI controllers.
+ * @orom: OROM struct buffer to fill out.
+ *
+ * Find OROM in UEFI ACPI table based on requested controllers.
+ * If found, the buffer with the OROM structure will be filled.
+ *
+ * Return: %MDADM_STATUS_SUCCESS on success, %MDADM_STATUS_ERROR otherwise.
+ */
+mdadm_status_t
+find_orom_in_acpi_uefi_tables(uefi_controller_t *uefi_controllers, int uefi_controllers_number,
```

Comment:
I miss explanation about unusual placement of it in ACPI tables. You have to clarify that it can be found in one of Intel provided `UEFI` ACPI table.